lib/graphql/schema/rescue_middleware.rb

```diff
@@ -0,0 +1,53 @@
+module GraphQL
+  class Schema
+    # - Store a table of errors & handlers
+    # - Rescue errors in a middleware chain, then check for a handler
+    # - If a handler is found, use it & return a {GraphQL::ExecutionError}
+    # - If no handler is found, re-raise the error
+    class RescueMiddleware
+      # @return [Hash] `{class => proc}` pairs for handling errors
+      attr_reader :rescue_table
+      def initialize
+        @rescue_table = {}
```

Comment:
Yeah, this is very much the simplest-thing-that-could-possibly-work. The implementation is private enough that I think it would be straightforward to swap out the Hash + `[]` for an Array + `.is_a?`